clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2450,6 +2450,26 @@ static bool BuiltinCountZeroBitsGeneric(Sema &S, CallExpr *TheCall) {
   return false;
 }
 
+// The argument must be a class or struct with a member
+// named type.
+static bool CheckBuiltinSyclKernelName(Sema &S, CallExpr *TheCall) {
+  QualType ArgTy = TheCall->getArg(0)->getType();
+  const auto *RT = ArgTy->getAs<RecordType>();
+
+  if(!RT)
+    return true;
+
+  RecordDecl *RD = RT->getDecl();
+  IdentifierTable &IdentTable = S.Context.Idents;
+  auto Name = DeclarationName(&(IdentTable.get("type")));
```

Comment:
I don't know if I should be worried about this, but `IdentifierTable::get()` will create an `IdentifierInfo` object if one doesn't already exist. That makes me nervous. There is a `find()` member function to do a lookup without creating a new object, but it isn't particularly ergonomic to use. The lack of a `getExisting()` or similar function that can return null is a little surprising to me though, so maybe I'm being overly paranoid. Maybe it is worth using `find()` here to avoid unlikely potential problems?